### PR TITLE
refactor: Port renderer/init to TypeScript

### PR DESCRIPTION
**Pull request by felixrieseberg**
Another chunk of TypeScript conversion, this time around for `renderer/init`.

Notes: no-notes

### CODE REVIEW

File: lib/renderer/init.ts [hunk 1]
```diff
@@ -34,7 +33,9 @@ webFrameInit()
 // Process command line arguments.
 const { hasSwitch, getSwitchValue } = process.atomBinding('command_line')
 
-const parseOption = function (name, defaultValue, converter = value => value) {
+const parseOption = function (
+  name: string, defaultValue: any, converter: (value: any) => any = value => value
```
Comment (MarshallOfSound) [inline lib/renderer/init.ts]: The `converter` here can be typed as `string` and `T`.

```suggestion
  name: string, defaultValue: T, converter: (value: string) => T = value => value
```

Then `parseOption` is strongly typed

File: typings/internal-ambient.d.ts [hunk 2]
```diff
@@ -36,7 +36,16 @@ declare namespace NodeJS {
   }
 }
 
+declare module NodeJS  {
```
Comment (MarshallOfSound) [inline typings/internal-ambient.d.ts]: This concerns me, why are we having to redeclare node globals, shouldn't they be coming through from `@types/node`

Comment (felixrieseberg) [reply typings/internal-ambient.d.ts]: Ah, I can answer that! `@types/node` has those defined as _actual_ globals.

Not defined:
```
global.__dirname
```

Defined:
```
__dirname
```

File: lib/renderer/init.ts (hunk 1 above)
Comment (felixrieseberg) [reply lib/renderer/init.ts]: Upon closer investigation, that doesn't work – and it makes sense, this here doesn't work: `(value: string) => T = value => value`

**Comment by felixrieseberg** [comment]
The two failures are unrelated :v:

**Pull request merged**